### PR TITLE
Provide examples of using `setup()` in Book and examples

### PR DESCRIPTION
**Pull request by ldesgoui**
Hello,

This adds a few examples to the setup chapter on what should be (?) the preferred way of adding Components and Resources, it also updates the examples to follow that methodology

Closes issue #482 

Thanks

---
This change is [Reviewable](https://reviewable.io/reviews/slide-rs/specs/487)

### CODE REVIEW

**Comment by ldesgoui** [comment]
I noted that AsyncDispatcher does not have a `setup()` function, should I attempt to add it in this PR ?
https://docs.rs/specs/0.12.3/specs/struct.AsyncDispatcher.html

**Comment by torkleyy** [comment]
Oh, not sure why that was left out.

File: book/src/07_setup.md [hunk 1]
```diff
@@ -81,6 +142,22 @@ impl<'a> System<'a> for Sys {
 This is much better; we can now use `setup` to fully initialize `Sys` without
 requiring our users to create and add resources manually to `World`!
 
-**If we override the `setup` function on a `System`, it is vitally important that we 
+**If we override the `setup` function on a `System`, it is vitally important that we
 remember to add `Self::SystemData::setup(res);`, or setup will not be performed for
-the `System`s `SystemData`.**
+the `System`s `SystemData`.** This usually causes panics during setup or during
```
[the remark this answers is not in the document]
Comment (ldesgoui) [reply book/src/07_setup.md]: I think what I meant is that the first manifestation of not including that would be noticeable directly (during setup or first dispatch), should I try rephrasing it differently ?

File: book/src/07_setup.md [hunk 2]
```diff
@@ -144,19 +144,17 @@ requiring our users to create and add resources manually to `World`!
...
 the first dispatch.
 
 ## Setting up in bulk
 
-In the case of libraries making use of `specs`, it is sometimes helpful to provide
-a way to add many Systems, Components and/or Resources at once. While `specs` could
-have something like a trait to generalize that, it is probably easier and more helpful
-for libraries themselves to create their own way. An example would be to simply have
-a standalone function:
+It's generally recommended to provide a standalone function to register multiple
```
Comment (torkleyy) [inline book/src/07_setup.md]: Oh, you could have left the first sentence. Sorry this could have been clearer.

**Pull request merged**